nixos/modules/services/audio/wyoming/faster-whisper.nix

```diff
@@ -160,7 +161,6 @@ in
           DevicePolicy = "closed";
           LockPersonality = true;
           MemoryDenyWriteExecute = true;
-          PrivateDevices = true;
```

Comment:
Alright, so this was about some of `/dev` not being visible to the process. It would be nice to figure out what it is we're actually missing in `DeviceAllow`.

Maybe instead of `strace ... |& grep /dev/nvidia` we do `strace ... |& grep /dev/`. We could just try editing https://github.com/NixOS/nixpkgs/blob/23a5f1f80dbc198c9a81ed0bea13f6be3b4b3689/nixos/modules/services/audio/wyoming/faster-whisper.nix#L137 directly, but we'd probably need to ughh tell systemd to allow the service to do `strace`. I presume it's something like `CapabilityBoundingSet = [ "CAP_SYS_PTRACE" ]`. Might be easier to just remove all of the sandboxing except for `PrivateDevices` and `DeviceAllow`

Reply:
Please restore this line.

> Might be easier to just remove all of the sandboxing except for PrivateDevices and DeviceAllow

Totally, I'm not a fan of polluting the runtime environment with debug flags that tie into strace and capabilities.

Reply:
So, I should restore the `PrivateDevices` line? What sandboxing should be removed?

Reply:
The `PrivateDevices` option populates `/dev` with only dummy devices.

Then `DeviceAllow` comes in, and I think it mounts the mentioned devices through into that private `/dev` environment.

Ideally no sandboxing would be removed, expect that which is required for CUDA, and then conditionally on CUDA usage.

Reply:
Iirc, even with PrivateDevices removed, it didn't work until I had `/dev/nvidia0` in DeviceAllow

Reply:
Are you saying that CUDA requires disabling `PrivateDevices=`, even with `/dev/nvidia0` in `DeviceAllow=`?

Reply:
Reading [here](https://www.freedesktop.org/software/systemd/man/latest/systemd.resource-control.html), it looks like PrivateDevices blocks access to all physical devices, but if the DevicePolicy is set to strict, then only devices in DeviceAllow are passed through. Maybe I'm misunderstanding, but that's what I saw. Let me know what you think.

Reply:
Ah, okay. Then let's drop it, I think `DevicePolicy=closed` is sufficient.